src/main/java/com/mraof/minestuck/client/particle/MSParticleType.java

```diff
@@ -0,0 +1,32 @@
+package com.mraof.minestuck.client.particle;
+
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.particle.ParticleEngine;
+import net.minecraft.core.particles.ParticleType;
+import net.minecraft.core.particles.SimpleParticleType;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import net.minecraftforge.client.event.RegisterParticleProvidersEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.ForgeRegistries;
+import net.minecraftforge.registries.RegistryObject;
+
+@Mod.EventBusSubscriber(modid = Minestuck.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
+public class MSParticleType
+{
+	public static final DeferredRegister<ParticleType<?>> REGISTER = DeferredRegister.create(ForgeRegistries.PARTICLE_TYPES, Minestuck.MOD_ID);
+	
+	public static final RegistryObject<SimpleParticleType> TRANSPORTALIZER = REGISTER.register("transportalizer", () -> new SimpleParticleType(false));
+	
+	@OnlyIn(Dist.CLIENT)
```

Comment:
This should generally be avoided in mod code. I recommend separating client-only code (particle provider registration) from common code (the register and transportalizer particle type), so that the common code isn't stored in the client package.

Reply:
Should I move MSParticleTypes to a different package? If so what should be left in the client package?

Reply:
Yeah, that'd be good. Since the particle provider should only be registered client-side, that's what should remain in the client package. If you want, you could try putting that function in `ClientProxy` until it grows large enough to warrant its own class.

Reply:
Ended up creating a new package, let me know if you find how its organized now suitable

Reply:
It's kind of the opposite problem with `TransportalizerParticle` now. Since I doubt that the common side of particles won't grow beyond `MSParticleTypes`, it might be better to put that specific class in util instead, and make a particle package in client for `TransportalizerParticle` (and future particle classes of that kind)

Reply:
Let me know if I got it how you want it!